### PR TITLE
Add jobs for PHP `8.3` to the CI

### PR DESCRIPTION
**Pull request by ste93cry**
With PHP `8.3` coming in November 2023, it's time to run the CI for that version as well so that we're ready for its release.

### CODE REVIEW

File: .github/workflows/static-analysis.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: '8.1'
+          php-version: '8.3'
```
Comment (Jean85) [inline .github/workflows/static-analysis.yaml:22]: This is failing, and PHP-CS-Fixer is slow to support newer language versions. We should stick to 8.2 here
```suggestion
          php-version: '8.2'
```

**Review by Jean85 — CHANGES_REQUESTED** [review]

File: .github/workflows/static-analysis.yaml (hunk 1 above)
Comment (ste93cry) [reply .github/workflows/static-analysis.yaml:22]: >PHP-CS-Fixer is slow to support newer language versions

It makes me sad, but I agree. I wanted to run the tol on PHP `8.3` just for the sake of consistency with the other jobs, but I remember that in the past we had to wait on average a long time for a compatible version to be released

**Comment by ste93cry** [comment]
@cleptric I think it's pointless to retest all PHP versions just to make sure they work without optional packages installed, so I removed some jobs from the GitHub Workflow and changed the matrix so that only the oldest and newest PHP versions of each major are tested. However, as you can see from the following image, some jobs are still flagged as required checks for the CI to pass. Maybe the repo settings need to be adjusted accordingly?

immagine

**Comment by cleptric** [comment]
The required checks haven been removed.

**Review by Jean85 — APPROVED** [review]

**Pull request merged**